analytics-api-core/src/main/scala/org/ekstep/analytics/api/service/JobAPIService.scala

```diff
@@ -76,42 +79,103 @@ class JobAPIService @Inject()(postgresDBUtil: PostgresDBUtil) extends Actor  {
+      else {
+        val computedFrom = if (dates.nonEmpty) Option(dates.get.from) else if (date.nonEmpty) date else from
+        val computedTo =  if (dates.nonEmpty) Option(dates.get.to) else if (date.nonEmpty) date else to
+
+        val objectLists = getExhaustObjectKeys(None, datasetId, computedFrom, computedTo, since, true)
+        val isValid = objectLists._1
+        val listObjs = objectLists._2
+
+        if ("true".equalsIgnoreCase(isValid.getOrElse("status", "false"))) {
+          if (listObjs.size > 0) {
+            val periodWiseFiles = listObjs.asInstanceOf[List[(String, String)]].groupBy(_._1).mapValues(_.map(_._2))
+            CommonUtil.OK(APIIds.PUBLIC_TELEMETRY_EXHAUST, Map("files" -> listObjs.asInstanceOf[List[(String, String)]].map(_._2), "periodWiseFiles" -> periodWiseFiles))
+          } else {
+            CommonUtil.OK(APIIds.PUBLIC_TELEMETRY_EXHAUST, Map("message" -> "Files are not available for requested date. Might not yet generated. Please come back later"))
+          }
+        } else {
+          APILogger.log("Request Validation FAILED")
+          CommonUtil.errorResponse(APIIds.PUBLIC_TELEMETRY_EXHAUST, isValid.getOrElse("message", ""), ResponseCode.CLIENT_ERROR.toString)
+        }
+      }
+    }
+    else {
+      APILogger.log("Request Validation FAILED for invalid datasetId")
+      CommonUtil.errorResponse(APIIds.PUBLIC_TELEMETRY_EXHAUST, isDatasetValid.getOrElse("message", ""), ResponseCode.CLIENT_ERROR.toString)
+    }
+
+  }
+
+  private def getExhaustObjectKeys(channel: Option[String], datasetId: String, from: Option[String], to: Option[String], since: Option[String] = None, isPublic: Boolean = false)(implicit config: Config, fc: FrameworkContext): (Map[String, String], List[(String, String)]) = {
+
+    val fromDate = if (since.nonEmpty) since.get else if (from.nonEmpty) from.get else CommonUtil.getPreviousDay()
+    val toDate = if (to.nonEmpty) to.get else CommonUtil.getToday()
+
+    val isValid = _validateRequest(channel, datasetId, fromDate, toDate, isPublic)
+    if ("true".equalsIgnoreCase(isValid.getOrElse("status", "false"))) {
       val loadConfig = config.getObject(s"channel.data_exhaust.dataset").unwrapped()
       val datasetConfig = if (null != loadConfig.get(datasetId)) loadConfig.get(datasetId).asInstanceOf[java.util.Map[String, AnyRef]] else loadConfig.get("default").asInstanceOf[java.util.Map[String, AnyRef]]
       val bucket = datasetConfig.get("bucket").toString
       val basePrefix = datasetConfig.get("basePrefix").toString
-      val prefix = basePrefix + datasetId + "/" + channel + "/"
+      val prefix = if(channel.isDefined) basePrefix + datasetId + "/" + channel.get + "/" else basePrefix + datasetId + "/"
       APILogger.log("prefix: " + prefix)
 
       val storageKey = config.getString("storage.key.config")
       val storageSecret = config.getString("storage.secret.config")
       val storageService = fc.getStorageService(storageType, storageKey, storageSecret)
```

Comment:
Storage config for public and private will be different. Also refactor the code a little bit - looks complicated now

Reply:
Added different storage config for public & private.